src/components/TextField/SbTextField.vue

```diff
@@ -0,0 +1,131 @@
+<template>
+  <div class="sb-textfield">
+    <label v-if="label" :for="id" class="sb-textfield__label">{{
+      label
+    }}</label>
+    <div class="sb-textfield__inner">
+      <span
+        v-if="sideText && sideText.side == 'left'"
+        class="sb-textfield__inner__input--with-text-model-left"
+        >{{ sideText.text }}</span
+      >
+      <input
+        :id="id"
+        v-model="computedValue"
+        class="sb-textfield__inner__input"
+        :type="internalType"
+        :placeholder="placeholder"
+        :name="name"
+        :value="nativeValue"
+        :class="componentClasses"
+        :required="required"
+        :disabled="disabled"
+      />
+      <SbIcon
+        v-if="icon && (withIconLeft || withIconRight || error)"
+        :size="icon.size"
+        :name="icon.name"
+        :class="'sb-textfield__inner__sb-icon--' + icon.side"
+      />
+      <SbIcon
+        v-if="clearable"
+        size="small"
+        name="x-clear"
+        class="sb-textfield__inner__sb-icon--right"
+        @click="computedValue = null"
+      />
+      <SbIcon
+        v-if="icon && type == 'password'"
+        size="small"
+        :name="icon.name"
+        class="sb-textfield__inner__sb-icon--right"
+        @click="handleShowHidePassword"
+      />
+      <span
+        v-if="sideText && sideText.side == 'right'"
+        class="sb-textfield__inner__input--with-text-model-right"
+        >{{ sideText.text }}</span
+      >
+    </div>
+    <span v-if="error && errorMessage" class="sb-textfield__message--error">{{
+      errorMessage
+    }}</span>
+  </div>
+</template>
+
+<script>
+import SbIcon from '../Icon'
+import TextfieldMixin from '../../mixins/textfield-mixin'
+
+export default {
+  name: 'SbTextField',
+
+  components: { SbIcon },
+
+  mixins: [TextfieldMixin],
+
+  computed: {
```

Comment:
I believe it's the code is not readable. You can refactor the computed classes to the following:

```js
hasSpecialClass() {
  return [
    this.error && 'sb-textfield__inner__input--error',
    this.ghost && 'sb-textfield__inner__input--ghost',
    !this.error && !this.ghost && 'sb-textfield__inner__input--default',
  ]
},
hasIcon() {
  return [
    this.withIconLeft && 'sb-textfield__inner__input--with-icon-left',
    this.withIconRight && 'sb-textfield__inner__input--with-icon-right',
  ]
},
hasTextOnSide() {
  return [
    this.withTextLeft && 'sb-textfield__inner__input--with-text-left',
    this.withTextRight && 'sb-textfield__inner__input--with-text-right',
  ]
},
componentClasses() {
  return [...this.hasTextOnSide, ...this.hasIcon, ...this.hasSpecialClass]
},
```

And, you can have just only the `componentClasses` computed that returns an array with all computed classes. What do you think?